docs/git-compatibility.md

```diff
@@ -109,6 +93,62 @@ https://github.com/octocat/Hello-World` will clone GitHub's "Hello-World" repo
 into a directory by the same name.
 
 
+## Co-located Jujutsu/Git repos
+
+A "co-located" Jujutsu repo is a hybrid Jujutsu/Git repo. These can be created
+if you initialize the Jujutsu repo in an existing Git repo by running `jj init
+--git-repo=.` or with `jj git clone --colocate`. The Git repo and the Jujutsu
+repo then share the same working copy. Jujutsu will import and export from and
+to the Git repo on every `jj` command automatically.
+
+This mode is very convenient useful when tools (e.g. build tools) expect a Git
+repo to be present.
+
+It is allowed to mix `jj` and `git` commands in such a repo in any order.
+However, it may be easier to keep track of what is going on if you mostly use
+read-only `git` commands and use `jj` to make changes to the repo. If you need
+to use several mutating `git` commands in a row, you may have to first `git
+switch` to a branch, and it may be easier to avoid using `jj` commands until the
+task you needed `git` for is finished.
+
+You can undo the results of mutating `git` commands using `jj undo` and `jj op
+restore`. Inside `jj op log`, changes by `git` will be represented as an "import
+git refs" operation.
+
+There are a few downsides to this mode of operation. Generally, using co-located
+repos may require you to deal with more involved Jujutsu and Git concepts.
+
+* Interleaving `jj` and `git` commands increases the chance of confusing branch
+  conflicts or [conflicted (AKA divergent) change
+  ids](glossary.md#divergent-change). These never lose data, but can be
+  annoying.
+
+* Git tools will have trouble with revisions that contain conflicted files. While
+  `jj` renders these files with conflict markers in the working copy, they are
+  stored in a non-human-readable fashion inside the repo. Git tools will often
+  see this non-human-readable representation.
+
+* When a `jj` branch is conflicted, the position of the branch in the Git repo
+  will disagree with one or more of the conflicted positions. The state of that
+  branch in git will be labeled as though it belongs to a remote named "git",
+  e.g. `branch@git`.
+
+* Jujutsu will ignore Git's staging area. It will not understand merge conflicts
+  as Git represents them, unfinished `git rebase` states, as well as other less
+  common states a Git repository can be in.
+
+* Colocated repositories are less resilient to
+  [concurrency](technical/concurrency.md#syncing-with-rsync-nfs-dropbox-etc)
+  issues if you share the repo using an NFS filesystem or Dropbox. In general,
+  such use of Jujutsu is not currently thoroughly tested.
```

Comment:
I don't think the "rsync problem" is specific to colocated setup. Our op log is resilient to concurrent operations, but the refs stored in the backing Git repo isn't. Our git metadata isn't either.

I think the difference is that `git import`/`export` is automated under the colocated setup, and we're more likely to see the problem out of the context.

Reply:
Oh, I forgot about the git metadata issues. This makes any kind of Dropbox-synced repo dangerous, since IIRC it causes a difficult-to-recover-from corrupt repo. I should at least link to the relevant bug from `concurrency.md`.

> I think the difference is that git import/export is automated under the colocated setup, and we're more likely to see the problem out of the context.

That's what I meant here, let me know if I could have said it better. I'll think about whether I want to rewrite something in light of the metadata issue as well.

Reply:
> Oh, I forgot about the git metadata issues. This makes any kind (co-located or not) of Dropbox-synced repo dangerous, since IIRC it causes a difficult-to-recover-from corrupt repo. 

Recovery (`jj debug reindex`) is relatively easy if you can notice the problem. Anyway, the problem would occur only if one side has authored change id, and the other side pulled that commit and has generated change id.

Reply:
I added a reference to #2193. I think that about covers it.